### PR TITLE
Alerting: Fix group filter

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR fixes group filter not working after this [PR](https://github.com/grafana/grafana/pull/73844) was merged. 
It also adds some warnings and add some examples in the help info tooltip about when the label matcher syntax is not correct in the filter.
For example, it's not clear that you cannot use spaces between labels and operators. With this change, user has a clue that something is wrong and can take a look to the examples.

**Special notes for your reviewer:**

https://github.com/grafana/grafana/assets/33540275/784a71a4-c2f3-4611-9656-1040e26dbb8e

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx [hunk 1]
```diff
@@ -30,42 +30,58 @@ export const MatcherFilter = ({ className, onFilterChange, defaultQueryString }:
   useEffect(() => onSearchInputChanged.cancel(), [onSearchInputChanged]);
 
   const searchIcon = <Icon name={'search'} />;
+  const inputInvalid =
+    defaultQueryString && defaultQueryString.length > 3 ? parseMatchers(defaultQueryString).length === 0 : false;
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx:34]: What's the reason for checking the length of the input here? 

The `{` and `}` are optional for the input.

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx:34]: > What's the reason for checking the length of the input here?

I was about to ask you as you added the same expression in this [PR](https://github.com/grafana/grafana/pull/61952/files#diff-bd5675bd3c94c3fc3286df05fd93b5cba420dd8a6b23c70db2b5e69800f4c96aR51)

I wanted to be consistent with this logic that is in different filters (Silences also use this expression)

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx:34]: > The { and } are optional for the input.

yes

Comment (gillesdemey) [reply public/app/features/alerting/unified/components/alert-groups/MatcherFilter.tsx]: Resolving this; it's probably been copied from various other places for a while and might have been there for historical reasons that are no longer relevant.

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**